### PR TITLE
Support exporting Tekton pipelines

### PR DESCRIPTION
**Pull request by ZhuGongpu**

### CODE REVIEW

**Comment by ZhuGongpu** [comment]
@Liujingfang1 @monopole @frankfarzan

File: internal/cmdexport/orchestrators/tekton.go [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+// limitations under the License.
+
+package orchestrators
+
+import (
+	"bytes"
+	"path"
+
+	"github.com/GoogleContainerTools/kpt/internal/cmdexport/types"
+	"sigs.k8s.io/kustomize/kyaml/yaml"
```
Comment (Liujingfang1) [inline internal/cmdexport/orchestrators/tekton.go:22]: Change it to
```
	"sigs.k8s.io/kustomize/kyaml/yaml"

	"github.com/GoogleContainerTools/kpt/internal/cmdexport/types"
```

File: internal/cmdexport/orchestrators/tekton_test.go [hunk 2]
```diff
@@ -0,0 +1,312 @@
...
+// limitations under the License.
+
+package orchestrators
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/GoogleContainerTools/kpt/internal/cmdexport/types"
+	"gotest.tools/assert"
```
Comment (Liujingfang1) [inline internal/cmdexport/orchestrators/tekton_test.go:22]: change it to
```
	"gotest.tools/assert"

    "github.com/GoogleContainerTools/kpt/internal/cmdexport/types"

```

File: internal/cmdexport/orchestrators/tekton.go [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+type TektonPipeline struct {
+	APIVersion string              `yaml:"apiVersion,omitempty"`
+	Kind       string              `yaml:",omitempty"`
+	Metadata   *TektonMetadata     `yaml:",omitempty"`
+	Spec       *TektonPipelineSpec `yaml:",omitempty"`
+	task       *TektonTask
+}
+
+func (p *TektonPipeline) Init(config *types.PipelineConfig) Pipeline {
+	p.APIVersion = "tekton.dev/v1beta1"
```
Comment (Liujingfang1) [inline internal/cmdexport/orchestrators/tekton.go:36]: Is `tekton.dev/v1beta1` the latest version?

File: internal/cmdexport/orchestrators/tekton.go [hunk 4]
```diff
@@ -0,0 +1,232 @@
...
+	p.Kind = "Pipeline"
+	p.Metadata = &TektonMetadata{Name: "run-kpt-functions"}
```
Comment (Liujingfang1) [inline internal/cmdexport/orchestrators/tekton.go:38]: It looks like "run-kpt-functions" are used in multiple files, can you change it to a constant?

File: internal/cmdexport/orchestrators/tekton.go [hunk 5]
```diff
@@ -0,0 +1,232 @@
...
+
+	b.Write(task)
+	b.WriteString("---\n")
+	b.Write(pipeline)
+
+	return b.Bytes(), nil
+}
+
+// TektonPipelineSpec describes the spec of a Pipeline.
+type TektonPipelineSpec struct {
```
Comment (Liujingfang1) [inline internal/cmdexport/orchestrators/tekton.go:94]: Can you change this to `PipelineSpec` for simplicity.

File: internal/cmdexport/orchestrators/tekton.go [hunk 6]
```diff
@@ -0,0 +1,232 @@
...
+	Workspaces []*TektonWorkspace    `yaml:",omitempty"`
+	Tasks      []*TektonPipelineTask `yaml:",omitempty"`
+}
+
+// // TektonPipelineTaskRef represents a task in a Tekton Pipeline object.
+type TektonPipelineTask struct {
```
Comment (Liujingfang1) [inline internal/cmdexport/orchestrators/tekton.go:100]: `TektonPipelineTask` to `PipelineTaskRef`.

File: internal/cmdexport/orchestrators/tekton.go [hunk 7]
```diff
@@ -0,0 +1,232 @@
...
+	return b.Bytes(), nil
+}
+
+// TektonPipelineSpec describes the spec of a Pipeline.
+type TektonPipelineSpec struct {
+	Workspaces []*TektonWorkspace    `yaml:",omitempty"`
+	Tasks      []*TektonPipelineTask `yaml:",omitempty"`
+}
+
+// // TektonPipelineTaskRef represents a task in a Tekton Pipeline object.
```
Comment (Liujingfang1) [inline internal/cmdexport/orchestrators/tekton.go:99]: Make sure the comment is consistent with the function name.

File: internal/cmdexport/orchestrators/tekton.go [hunk 8]
```diff
@@ -0,0 +1,232 @@
...
+type TektonPipelineTask struct {
+	Name       string                 `yaml:",omitempty"`
+	TaskRef    *TektonPipelineTaskRef `yaml:"taskRef,omitempty"`
+	RunAfter   []string               `yaml:"runAfter,omitempty"`
+	Workspaces []*TektonWorkspace     `yaml:",omitempty"`
+}
+
+// TektonPipelineTaskRef represents a taskRef field in a task of a Tekton Pipeline object.
+type TektonPipelineTaskRef struct {
```
Comment (Liujingfang1) [inline internal/cmdexport/orchestrators/tekton.go:108]: Change it to `TaskRef` for simplicity.

File: internal/cmdexport/orchestrators/tekton.go [hunk 9]
```diff
@@ -0,0 +1,232 @@
...
+	Name string `yaml:",omitempty"`
+}
+
+// TektonMetadata contains metadata to describe a resource object.
+type TektonMetadata struct {
```
Comment (Liujingfang1) [inline internal/cmdexport/orchestrators/tekton.go:113]: Suggest to rename the types for pipeline to start with `Pipeline` and the types for tasks to start with `Task`.

**Pull request merged**